gateleen-routing/src/main/java/org/swisspush/gateleen/routing/Forwarder.java

```diff
@@ -475,6 +475,11 @@ private Handler<AsyncResult<HttpClientResponse>> getAsyncHttpClientResponseHandl
             cRes.endHandler(v -> {
                 try {
                     req.response().end();
+
+                    // if everything is fine, we call the after handler
+                    if (afterHandler != null && cRes.statusCode() == StatusCode.OK.getStatusCode()) {
```

Comment:
Are you sure that the status code **200 OK** is enough? What happens when a client request goes into a queue? Then you will get a **202 Accepted** as the answer. Maybe there are other cases like **304 Not Modified** which are also OK to continue afterward.

Reply:
Good point, this was taken from

https://github.com/swisspost/gateleen/blob/13617adf3205fcc3badd110337bc6d67c68377ee/gateleen-hook/src/main/java/org/swisspush/gateleen/hook/HookHandler.java#L1203

so do we have the same problem there as well? Not sure if we have to prepare for a 304, we normally don't see this for a PUT / POST / DELETE?

Reply:
I agree that we have the same problem also in the HookHandler. **304 Not Modified** status codes are actually (at least in the gateleen environment) only used for GET requests where the Storage and EtagHandler are involved.

I would suggest checking for any **2xx** status codes. They should all be OK. What do you think?

Reply:
OK, let's do this